src/deno_json/mod.rs

```diff
@@ -553,6 +553,24 @@ impl ConfigFileReadError {
   }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum NodeModulesMode {
+  LocalAuto,
+  LocalManual,
+  GlobalAuto,
+}
+
+impl NodeModulesMode {
+  pub fn parse(s: &str) -> Result<Self, AnyError> {
+    match s {
+      "local-auto" => Ok(Self::LocalAuto),
+      "local-manual" => Ok(Self::LocalManual),
+      "global-auto" => Ok(Self::GlobalAuto),
+      s => bail!("Unsupported 'nodeModules' value '{}'. Supported: 'local-auto', 'local-manual', 'global-auto'.", s), 
```

Comment:
It would be good to get rid of the use of anyhow here and instead see `thiserror`. There are still some areas in this crate that still use anyhow, but they should be migrated to thiserror in the future.